[Pico] HID Mouse does not work on macOS Big Sur 11.2.1

**Firmware**

```python
Adafruit CircuitPython 6.2.0-beta.4 on 2021-03-18; Raspberry Pi Pico with rp2040
```

**Code/REPL**

```python
import time
import board
import digitalio
import analogio
import usb_hid
from adafruit_hid.mouse import Mouse
from adafruit_hid.keyboard import Keyboard
from adafruit_hid.keycode import Keycode
mouse = Mouse(usb_hid.devices)
kbd = Keyboard(usb_hid.devices)
x_axis = analogio.AnalogIn(board.A0)
y_axis = analogio.AnalogIn(board.A1)
btn_left = digitalio.DigitalInOut(board.GP22)
btn_left.direction = digitalio.Direction.INPUT
btn_left.pull = digitalio.Pull.UP
pot_min = 0.00
pot_max = 3.29
step = (pot_max - pot_min) / 20.0
def get_voltage(pin):
    return (pin.value * 3.3)/65536
def steps(axis):
    return round((axis - pot_min) / step)
while True:
    x = get_voltage(x_axis)
    y = get_voltage(y_axis)
    print(x, y, btn_left.value)
    if steps(x) > 11.0:
        print('x>11', steps(x))
        mouse.move(x=10)
    if steps(x) < 9.0:
        mouse.move(x=-10)
    if steps(y) > 11.0:
        print('y>11', steps(x))
        mouse.move(y=10)
    if steps(y) < 9.0:
        mouse.move(y=-10)    
    if not btn_left.value:
        print('button')
        mouse.click(Mouse.LEFT_BUTTON)
        time.sleep(0.2)
    time.sleep(0.1)
```

**Behavior**

The mouse does not work on macOS Big Sur 11.2.1 but it works correctly on Windows 10 and Lubuntu 20.04.1.

**Description**
The board has an analog joystick (two potentiometers and a tact switch) connected. They are wired and configured correctly as in the console output I can see the correct voltages and switch status logged. What is more, the code above works correctly on Windows 10 and Lubuntu 20.04.1 - it moves the mouse cursor and performs the mouse clicks.
On macOS Big Sur 11.2.1 it has no effect at all - the cursor does not move and the click event does not work.

I tried adding the following keyboard key press code instead of the mouse click and it does work correctly on the Mac.
```
kbd.press(Keycode.A)  # "Press"...
kbd.release_all()  # ..."Release"!
```

Please let me know if there is anything I can do to help debugging or fixing this issue.

I am trying to make a custom build but I keep getting errors. I followed all setup instructions from the documentation and I tested it on macOS Big Sur and Ubuntu 20.04.

On Ubuntu I get the following error:
`ubuntu@circuitpython:~/circuitpython/ports/raspberrypi$ make BOARD=raspberry_pi_pico
Use make V=1, make V=2 or set BUILD_VERBOSE similarly in your environment to increase build verbosity.
QSTR updated
Traceback (most recent call last):
  File "gen_stage2.py", line 82, in <module>
    typer.run(main)
  File "/home/ubuntu/.local/lib/python3.8/site-packages/typer/main.py", line 859, in run
    app()
  File "/home/ubuntu/.local/lib/python3.8/site-packages/typer/main.py", line 214, in __call__
    return get_command(self)(*args, **kwargs)
  File "/home/ubuntu/.local/lib/python3.8/site-packages/typer/main.py", line 239, in get_command
    click_command = get_command_from_info(typer_instance.registered_commands[0])
  File "/home/ubuntu/.local/lib/python3.8/site-packages/typer/main.py", line 425, in get_command_from_info
    command = cls(  # type: ignore
TypeError: __init__() got an unexpected keyword argument 'no_args_is_help'
make: *** [Makefile:271: build-raspberry_pi_pico/stage2.c] Error 1`

On the mac I get the following error:
`➜  raspberrypi git:(main) ✗ make BOARD=raspberry_pi_pico
Use make V=1, make V=2 or set BUILD_VERBOSE similarly in your environment to increase build verbosity.
Traceback (most recent call last):
  File "gen_stage2.py", line 2, in <module>
    import cascadetoml
ModuleNotFoundError: No module named 'cascadetoml'
make: *** [build-raspberry_pi_pico/stage2.c] Error 1`

Please let me know if this is a known issue. I will keep experimenting with the build (perhaps build one of the release branches/tags instead of master).

What is more, I would suggest updating the MacOS Setup guide to replace the `brew install gcc-arm-embedded` with the manual installation (similar to the Linux setup). That is because the `gcc-arm-none-eabi-10-2020-q4-major-mac.tar.bz2` package is not signed and the MacOS prevents executing the compiler binary with the "Unidentified Developer" error. The `gcc-arm-none-eabi-10-2020-q4-major-mac.pkg` package is "Signed and notarized" and it works fine.

--- a/tools/gen_usb_descriptor.py
+++ b/tools/gen_usb_descriptor.py
@@ -17,9 +17,11 @@ ALL_DEVICES = "CDC CDC2 MSC AUDIO HID VENDOR"
 ALL_DEVICES_SET = frozenset(ALL_DEVICES.split())
 DEFAULT_DEVICES = "CDC MSC AUDIO HID"
 
+# This list is in preferred order. MacOS does not like GAMEPAD coming before MOUSE.
 ALL_HID_DEVICES = (
     "KEYBOARD MOUSE CONSUMER SYS_CONTROL GAMEPAD DIGITIZER XAC_COMPATIBLE_GAMEPAD RAW"
 )
+ALL_HID_DEVICES_ORDER = dict((name, idx) for (idx, name) in enumerate(ALL_HID_DEVICES.split()))
 ALL_HID_DEVICES_SET = frozenset(ALL_HID_DEVICES.split())
 # Digitizer works on Linux but conflicts with mouse, so omit it.
 DEFAULT_HID_DEVICES = "KEYBOARD MOUSE CONSUMER GAMEPAD"
@@ -352,7 +354,8 @@ if include_hid:
     else:
         report_id = 1
         concatenated_descriptors = bytearray()
-        for name in args.hid_devices:
+        # Sort HID devices by preferred order.
+        for name in sorted(args.hid_devices, key=ALL_HID_DEVICES_ORDER.get):
             concatenated_descriptors.extend(
                 bytes(hid_report_descriptors.REPORT_DESCRIPTOR_FUNCTIONS[name](report_id))
             )

